cvmfs/swissknife_pull.cc

```diff
@@ -237,6 +239,16 @@ static bool Pull(const shash::Any &catalog_hash, const std::string &path) {
     return true;
   }
 
+  // Check if the catalog matches the pathfilter
+  if (path != ""              &&  // necessary to load the root catalog
+      pathfilter              &&
+      pathfilter->IsValid()   &&
```

Comment:
This could be checked when the `-d` option is parsed and, in case of invalid filter, the program can stop immediately.